Common/Brokerages/DefaultBrokerageModel.cs

```diff
@@ -236,5 +242,22 @@ public virtual ISettlementModel GetSettlementModel(Security security, AccountTyp
             return new ImmediateSettlementModel();
         }
 
+        protected bool ValidateQuantityFloored(decimal quantity, out BrokerageMessageEvent message)
+        {
+            message = null;
+
+            // validate quantity floored
+            if (quantity != Math.Floor(quantity))
```

Comment:
Yes, that's a better approach. I had also meant to add unit tests for the new method babbe81.